### PR TITLE
Error info extension and output fix

### PR DESCRIPTION
**Pull request by alexmaslenn**

### CODE REVIEW

**Comment by alexmaslenn** [comment]
Closes #234 
Closes #233

File: src/opera/threading/utils.py [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+class SafePrinter:
+    lock = Lock()
+
+    @classmethod
+    def print_lines(cls, wrapper: TextIOWrapper):
+        cls.lock.acquire()
+        print_thread("------------")
+        for line in wrapper:
+            print(line.rstrip())
+        cls.lock.release()
```
Comment (anzoman) [inline src/opera/threading/utils.py:18]: Sanity tests fail with `opera/threading/utils.py|14:8|SafePrinter.print_lines |R1732|consider-using-with|Consider using 'with' for resource-allocating operations`, so you could probably change these lines to:

```suggestion
        with cls.lock:
            print_thread("------------")
            for line in wrapper:
                print(line.rstrip())
```

**Review by anzoman — CHANGES_REQUESTED** [review]
Thanks for these fixes @alexmaslenn, it looks okay. I just left one comment regarding the failure of our sanity tests.

File: src/opera/threading/utils.py (hunk 1 above)
Comment (alexmaslenn) [reply src/opera/threading/utils.py]: Done.

**Review by anzoman — APPROVED** [review]
Looks good, @alexmaslenn thanks for your contribution!

**Pull request merged**